### DISCUSSION TITLE
Is possible micronucleus for ATtiny87 (as for Digispark Pro ATtiny167) ?

### DISCUSSION THREAD

**Discussion by apws** [SpenceKonde/ATTinyCore#632] [General]
It is enough to tweak boards.txt to have this chip and limit upload size?

**Comment by SpenceKonde** [comment]
That is on the list for next releaae

____________
Spence Konde
Azzy’S Electronics

New products! Check them out at tindie.com/stores/DrAzzy
GitHub: github.com/SpenceKonde
ATTinyCore: Arduino support for almost every ATTiny microcontroller
Contact: ***@***.***

**Reply by SpenceKonde** [reply]
and no, it;s not enough to tweak boards.txt, bootloader needs to be rebuilt. There is a greater degree of difference than you might expect, because the 87 is below the 8k barrier while the 167 is not: 
* The 87 can reach any location in it's memory with an rjmp instruction from any other location. 
* The 167 cannot. 
* Thus the 87 has no JMP instruction, and uses 2 byte vectors. The 167 does need JMP, (as well as CALL) and it's vector table uses 4 byte vectors. 
* Since one of the core functions of  virtualboot bootloaders like this is to rewrite vectors, that is a difference that you just can't ignore.

**Reply by apws** [reply]
ya, thanks for update :-) great work...
